kotlinpoet/src/test/java/com/squareup/kotlinpoet/KmSpecsTest.kt

```diff
@@ -27,17 +34,15 @@ class KmSpecsTest {
   @Test
   fun constructorData() {
     val typeSpec = ConstructorClass::class.toTypeSpec()
-
-    assertThat(typeSpec.primaryConstructor).isNotNull()
-    assertThat(typeSpec.primaryConstructor?.parameters).hasSize(2)
-    val fooParam = typeSpec.primaryConstructor!!.parameters[0]
-    assertThat(fooParam.name).isEqualTo("foo")
-    assertThat(fooParam.type).isEqualTo(String::class.asClassName())
-    assertThat(fooParam.modifiers).doesNotContain(KModifier.VARARG)
-    val barParam = typeSpec.primaryConstructor!!.parameters[1]
-    assertThat(barParam.name).isEqualTo("bar")
-    assertThat(barParam.modifiers).contains(KModifier.VARARG)
-    assertThat(barParam.type).isEqualTo(INT)
+    //language=kotlin
+    assertThat(typeSpec.trimmedToString()).isEqualTo("""
+      class ConstructorClass(
+        val foo: kotlin.String,
+        vararg bar: kotlin.Int
+      ) {
+        constructor(bar: kotlin.Int)
```

Comment:
Oh, so the secondary constructor isn't ignored as stated in the comment inside `ConstructorClass`?

Reply:
I think semantics are maybe mixed up. The comment is in reference to the `: this(...)` that the secondary constructor should call. Not sure how to read that from metadata